CoffeeChat/View/MeetupStatusView.swift

```diff
@@ -0,0 +1,335 @@
+        pearImageView.layer.shadowOffset = CGSize(width: 0.0, height: 2.0)
+        pearImageView.layer.shadowOpacity = 0.15
+        pearImageView.layer.shadowRadius = 2
+        backgroundShadingView.addSubview(pearImageView)
+    }
+
+    private func setupConstraints() {
+        statusImageView.snp.makeConstraints { make in
+            make.size.equalTo(statusImageSize)
+            make.leading.equalToSuperview()
+            make.centerY.equalTo(statusLabel)
+        }
+
+        statusLabel.snp.makeConstraints { make in
+            make.leading.equalTo(statusImageView.snp.trailing).offset(4)
+            make.top.equalToSuperview()
+        }
+
+        backgroundShadingView.snp.makeConstraints { make in
+            make.top.equalTo(statusLabel.snp.bottom).offset(statusMessagePadding)
+            make.leading.trailing.bottom.equalToSuperview()
+        }
+
+        pearImageView.snp.makeConstraints { make in
+            make.leading.top.equalToSuperview().inset(pearPadding)
+            make.size.equalTo(pearImageSize)
+        }
+
+        messageTextView.snp.makeConstraints { make in
+            make.leading.equalTo(pearImageView.snp.trailing).offset(messageLeadingPadding)
+            make.top.bottom.equalToSuperview().inset(messageVerticalPadding)
+            make.trailing.equalToSuperview().inset(messageTrailingPadding)
+        }
+    }
+
+    // MARK: Different View States
+
+    private func setupForResponding(to user: User) {
+        statusImageView.image = UIImage(named: "new-pear")
+        statusLabel.text = "New Pear"
+        messageTextView.attributedText = unformattedText(for: "\(user.firstName) wants to meet you! Pick a time that works for both of you.")
+    }
+
+    private func setupForWaiting(for user: User) {
+        statusImageView.image = UIImage(named: "reached-out")
+        statusLabel.text = "Reached out"
+        messageTextView.attributedText = unformattedText(for: "Just waiting on \(user.firstName) to pick a time and place!")
+    }
+
+    private func setupForChatScheduled(on date: Date, for user: User) {
+        statusImageView.image = UIImage(named: "scheduled")
+        statusLabel.text = "Chat scheduled"
+
+        let prefix = unformattedText(for: "Get pumped to meet on ")
+        let body = underlinedText(for: "\(formatDate(date))")
+        let suffix = unformattedText(for: ".")
+
+        let fullText = NSMutableAttributedString()
+        fullText.append(prefix)
+        fullText.append(body)
+        fullText.append(suffix)
+
+        if user.instagram != nil || user.facebook != nil {
+            let contactPrefix = unformattedText(for: "\nYou can reach \(user.firstName) on ")
+            let contactLinks = textForSocialMedia(facebook: user.facebook, instagram: user.instagram)
+            let contactSuffix = unformattedText(for: ".")
+            fullText.append(contactPrefix)
+            fullText.append(contactLinks)
+            fullText.append(contactSuffix)
+        }
```

Comment:
I'm not sure if this is 100% relevant but I did something like this for another project before where I had to combine many different styles into one string. Just thought I'd include it here in case you want to take a look. It might take more effort for you to refactor and use this convention tho... So we can also maybe keep this in mind if you think it can work and would be cleaner and we can open another issue to change this later? 

So like in my extension, I add the different styles I need.
```
extension NSMutableAttributedString {
     func boldFont14(_ value:String) -> NSMutableAttributedString {
         let attributes: [NSAttributedString.Key : UIFont] = [
         .font : .boldSystemFont(ofSize: 14)
         ]
         self.append(NSAttributedString(string: value, attributes:attributes))
         return self
     }

      func normalFont14(_ value:String) -> NSMutableAttributedString {
         let attributes: [NSAttributedString.Key : UIFont] = [
             .font : .systemFont(ofSize: 14),
         ]
         self.append(NSAttributedString(string: value, attributes:attributes))
         return self
     }
 }
```

Then later when I have to make the text, I do
```
private func setupFriendAcceptCell(fromUser: String) {
         notificationLabel.attributedText =
             NSMutableAttributedString()
                 .boldFont14(fromUser)
                 .normalFont14(" accepted your friend request ")
         layoutNotificationLabel()
     }
```

Reply:
I really like this